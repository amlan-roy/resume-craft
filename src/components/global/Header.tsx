"use client";

import React from "react";
import { useTheme } from "next-themes";
import { Label } from "@/components/ui/label";
import { Switch } from "@/components/ui/switch";
import Logo from "@/components/global/Logo";

type HeaderProps = {};

const Header: React.FC<HeaderProps> = () => {
  const { setTheme, theme } = useTheme();
  return (
    <header className="p-4 w-full flex justify-between mx-auto max-w-screen-xl">
      <Logo linkify />
      <div className="flex items-center space-x-2 mx-2">
        <Switch
          id="dark-mode"
          checked={theme === "dark"}
          onCheckedChange={() => {
            theme === "dark" ? setTheme("light") : setTheme("dark");
          }}
        />
        <Label htmlFor="dark-mode">Dark Mode</Label>
      </div>
    </header>
  );
};
export default Header;
