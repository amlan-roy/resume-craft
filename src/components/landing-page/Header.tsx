"use client";
import React from "react";
import Logo from "@/components/global/Logo";
import { Button } from "../ui/button";
import { useTheme } from "next-themes";
import { Switch } from "@/components/ui/switch";
import { Label } from "@/components/ui/label";
import Link from "next/link";

type HeaderProps = {};

const Header: React.FC<HeaderProps> = () => {
  const { setTheme, theme } = useTheme();
  return (
    <header data-testid="landing-page__header" className="p-4 w-full flex">
      <div className="flex justify-between items-center w-full max-w-2xl mx-auto">
        <Logo linkify />
        <div className="sm:flex gap-3 hidden">
          {/* <Link href={"/generate-resume"} passHref> */}
          <Button variant={"outline"} disabled title="Coming soon">
            Generate Resume
          </Button>
          {/* </Link> */}
          <Link href={"/generate-variants"} passHref>
            <Button>Generate Variants</Button>
          </Link>
        </div>
      </div>
      <div className="flex items-center space-x-2 mx-2">
        <Switch
          id="dark-mode"
          checked={theme === "dark"}
          onCheckedChange={() => {
            theme === "dark" ? setTheme("light") : setTheme("dark");
          }}
        />
        <Label htmlFor="dark-mode">Dark Mode</Label>
      </div>
    </header>
  );
};
export default Header;
