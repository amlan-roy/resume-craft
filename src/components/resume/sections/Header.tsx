import React from "react";
import { Document, Page, StyleSheet, Text, View } from "@react-pdf/renderer";
import * as z from "zod";
import { basicDetailsSectionSchema } from "@/lib/types/form";

type HeaderProps = {
  data: z.infer<typeof basicDetailsSectionSchema>;
};

const Header: React.FC<HeaderProps> = ({ data }) => {
  const {
    fields: {
      name,
      email,
      phone,
      location,
      portfolioUrl,
      githubUrl,
      linkedinUrl,
      stackoverflowUrl,
      blogUrl,
    },
  } = data;

  const additionalInfo = [
    location,
    phone,
    email,
    portfolioUrl,
    githubUrl,
    linkedinUrl,
    stackoverflowUrl,
    blogUrl,
  ].filter((item) => item);
  return (
    <View style={styles.container}>
      <Text style={styles.name}>{name}</Text>
      <View style={styles.additionalInfo}>
        {additionalInfo.map((item, index) => {
          let Item = <></>;
          if (item) {
            switch (item) {
              case location:
                Item = (
                  <View
                    key={index}
                    style={{
                      display: "flex",
                      alignItems: "center",
                      gap: "2rem",
                    }}
                  >
                    <Text>{location}</Text>
                  </View>
                );
                break;
              case phone:
                Item = (
                  <View
                    key={index}
                    style={{
                      display: "flex",
                      alignItems: "center",
                      gap: "2rem",
                    }}
                  >
                    <Text>{phone}</Text>
                  </View>
                );
                break;
              case email:
                Item = (
                  <View
                    key={index}
                    style={{
                      display: "flex",
                      alignItems: "center",
                      gap: "2rem",
                      color: "#1D4ED8",
                    }}
                  >
                    <a href={`mailto:${email}`}>{email}</a>
                  </View>
                );
                break;
              case portfolioUrl:
                Item = (
                  <View
                    key={index}
                    style={{
                      display: "flex",
                      alignItems: "center",
                      gap: "2rem",
                      color: "#1D4ED8",
                    }}
                  >
                    <a href={portfolioUrl}>{portfolioUrl}</a>
                  </View>
                );
                break;
              case githubUrl:
                Item = (
                  <View
                    key={index}
                    style={{
                      display: "flex",
                      alignItems: "center",
                      gap: "2rem",
                      color: "#1D4ED8",
                    }}
                  >
                    <a href={githubUrl}>{githubUrl}</a>
                  </View>
                );
                break;
              case linkedinUrl:
                Item = (
                  <View
                    key={index}
                    style={{
                      display: "flex",
                      alignItems: "center",
                      gap: "2rem",
                      color: "#1D4ED8",
                    }}
                  >
                    <a href={linkedinUrl}>{linkedinUrl}</a>
                  </View>
                );
                break;
              case stackoverflowUrl:
                Item = (
                  <View
                    key={index}
                    style={{
                      display: "flex",
                      alignItems: "center",
                      gap: "2rem",
                      color: "#1D4ED8",
                    }}
                  >
                    <a href={stackoverflowUrl}>{stackoverflowUrl}</a>
                  </View>
                );
                break;
              case blogUrl:
                Item = (
                  <View
                    key={index}
                    style={{
                      display: "flex",
                      alignItems: "center",
                      gap: "2rem",
                      color: "#1D4ED8",
                    }}
                  >
                    <a href={blogUrl}>{blogUrl}</a>
                  </View>
                );
                break;
              default:
                Item = <View key={index}></View>;
                break;
            }
          }
          return (
            <>
              {Item}
              {index < additionalInfo.length - 1 ? <Text>{" | "}</Text> : <></>}
            </>
          );
        })}
      </View>
    </View>
  );
};
export default Header;

const styles = StyleSheet.create({
  container: {
    width: "100%",
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
    justifyContent: "center",
  },
  name: {
    fontSize: 16,
    fontWeight: "bold",
    color: "#404040",
    textAlign: "center",
    fontFamily: "Helvetica-Bold",
  },
  additionalInfo: {
    display: "flex",
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "center",
    padding: 8,
    fontSize: 11,
  },
});
