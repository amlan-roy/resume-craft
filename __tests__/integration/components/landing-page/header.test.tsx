import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import Header from "@/components/landing-page/Header";

describe("integration:components/landing-page/header", () => {
  it("snapshot", () => {
    const tree = render(<Header />);
    expect(tree.container).toMatchSnapshot();
  });

  //Todo: Had issue mocking the next themes and working with ThemeProvider here. Add proper tests later
  it("Header renders initial theme correctly", () => {
    render(<Header />);
    expect(screen.getByTestId("landing-page__header")).toBeInTheDocument();
    expect(screen.getByLabelText("Dark Mode")).toBeInTheDocument();
    expect(
      screen.getByRole("switch", { name: /Dark Mode/i })
    ).not.toBeChecked();
  });
});
