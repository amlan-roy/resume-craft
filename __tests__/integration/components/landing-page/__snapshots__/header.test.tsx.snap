// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`integration:components/global/header snapshot 1`] = `
<div>
  <header
    class="p-4 w-full flex justify-between mx-auto max-w-screen-xl"
    data-testid="global__header"
  >
    <a
      class="flex justify-center items-center"
      data-testid="global__logo-linkified"
      href="/"
    >
      <img
        alt="Resume Craft Icon"
        class="w-9 h-9 sm:w-full sm:h-full"
        data-nimg="1"
        decoding="async"
        height="36"
        loading="lazy"
        src="/_next/image?url=%2Ficon.png&w=96&q=75"
        srcset="/_next/image?url=%2Ficon.png&w=48&q=75 1x, /_next/image?url=%2Ficon.png&w=96&q=75 2x"
        style="color: transparent;"
        width="36"
      />
      <div
        class="flex flex-col text-brand-neutral-12 mx-3 justify-center items-center"
      >
        <p
          class="text-base sm:text-xl font-extrabold leading-none sm:leading-none"
        >
          RESUME
        </p>
        <p
          class="text-base sm:text-xl leading-none sm:leading-none "
        >
          <span
            class="font-light"
          >
             - 
          </span>
          craft
          <span
            class="font-light"
          >
             - 
          </span>
        </p>
      </div>
    </a>
    <div
      class="flex items-center space-x-2 mx-2"
    >
      <button
        aria-checked="false"
        class="peer inline-flex h-6 w-11 shrink-0 cursor-pointer items-center rounded-full border-2 border-transparent transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 focus-visible:ring-offset-background disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=unchecked]:bg-input"
        data-state="unchecked"
        id="dark-mode"
        role="switch"
        type="button"
        value="on"
      >
        <span
          class="pointer-events-none block h-5 w-5 rounded-full bg-background shadow-lg ring-0 transition-transform data-[state=checked]:translate-x-5 data-[state=unchecked]:translate-x-0"
          data-state="unchecked"
        />
      </button>
      <label
        class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
        for="dark-mode"
      >
        Dark Mode
      </label>
    </div>
  </header>
</div>
`;

exports[`integration:components/landing-page/header snapshot 1`] = `
<div>
  <header
    class="p-4 w-full flex"
    data-testid="landing-page__header"
  >
    <div
      class="flex justify-between items-center w-full max-w-2xl mx-auto"
    >
      <a
        class="flex justify-center items-center"
        data-testid="global__logo-linkified"
        href="/"
      >
        <img
          alt="Resume Craft Icon"
          class="w-9 h-9 sm:w-full sm:h-full"
          data-nimg="1"
          decoding="async"
          height="36"
          loading="lazy"
          src="/_next/image?url=%2Ficon.png&w=96&q=75"
          srcset="/_next/image?url=%2Ficon.png&w=48&q=75 1x, /_next/image?url=%2Ficon.png&w=96&q=75 2x"
          style="color: transparent;"
          width="36"
        />
        <div
          class="flex flex-col text-brand-neutral-12 mx-3 justify-center items-center"
        >
          <p
            class="text-base sm:text-xl font-extrabold leading-none sm:leading-none"
          >
            RESUME
          </p>
          <p
            class="text-base sm:text-xl leading-none sm:leading-none "
          >
            <span
              class="font-light"
            >
               - 
            </span>
            craft
            <span
              class="font-light"
            >
               - 
            </span>
          </p>
        </div>
      </a>
      <div
        class="sm:flex gap-3 hidden"
      >
        <button
          class="inline-flex items-center justify-center whitespace-nowrap rounded-full text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border-2 bg-transparent text-brand-secondary-blue-10 border-brand-secondary-blue-10 hover:border-brand-secondary-blue-8 active:border-brand-secondary-blue-6 disabled:border-brand-neutral-7 disabled:text-brand-neutral-8 dark:border-brand-secondary-blue-2 dark:hover:border-brand-secondary-blue-4 dark:active:border-brand-secondary-blue-6 dark:disabled:border-brand-neutral-7 dark:disabled:text-brand-neutral-6 h-10 px-4 py-2"
          disabled=""
          title="Coming soon"
        >
          Generate Resume
        </button>
        <a
          href="/generate-variants"
        >
          <button
            class="inline-flex items-center justify-center whitespace-nowrap rounded-full text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-brand-secondary-blue-4 text-brand-secondary-blue-10 hover:bg-brand-secondary-blue-6 active:bg-brand-secondary-blue-8 disabled:bg-brand-primary-green-7 disabled:text-brand-neutral-8 dark:bg-brand-secondary-blue-4 dark:text-brand-secondary-blue-10 dark:hover:bg-brand-secondary-blue-6 dark:active:bg-brand-secondary-blue-8 dark:disabled:bg-brand-neutral-7 dark:disabled:text-brand-neutral-8 h-10 px-4 py-2"
          >
            Generate Variants
          </button>
        </a>
      </div>
    </div>
    <div
      class="flex items-center space-x-2 mx-2"
    >
      <button
        aria-checked="false"
        class="peer inline-flex h-6 w-11 shrink-0 cursor-pointer items-center rounded-full border-2 border-transparent transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 focus-visible:ring-offset-background disabled:cursor-not-allowed disabled:opacity-50 data-[state=checked]:bg-primary data-[state=unchecked]:bg-input"
        data-state="unchecked"
        id="dark-mode"
        role="switch"
        type="button"
        value="on"
      >
        <span
          class="pointer-events-none block h-5 w-5 rounded-full bg-background shadow-lg ring-0 transition-transform data-[state=checked]:translate-x-5 data-[state=unchecked]:translate-x-0"
          data-state="unchecked"
        />
      </button>
      <label
        class="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
        for="dark-mode"
      >
        Dark Mode
      </label>
    </div>
  </header>
</div>
`;
